src/planner.c

```diff
@@ -489,22 +493,77 @@ should_optimize_query()
 
 extern void sort_transform_optimization(PlannerInfo *root, RelOptInfo *rel);
 
+static inline bool
+should_optimize_append(const Path *path)
+{
+	RelOptInfo *rel = path->parent;
+	ListCell   *lc;
+
+	if (!guc_constraint_aware_append ||
+		constraint_exclusion == CONSTRAINT_EXCLUSION_OFF)
+		return FALSE;
+
+	/*
+	 * If there are clauses that have mutable functions, this path is ripe for
+	 * execution-time optimization
+	 */
+	foreach(lc, rel->baserestrictinfo)
+	{
+		RestrictInfo *rinfo = (RestrictInfo *) lfirst(lc);
+
+		if (contain_mutable_functions((Node *) rinfo->clause))
+			return TRUE;
+	}
+	return FALSE;
+}
+
 static void
 timescaledb_set_rel_pathlist(PlannerInfo *root,
 							 RelOptInfo *rel,
 							 Index rti,
 							 RangeTblEntry *rte)
 {
-	if (extension_is_loaded() && !IS_DUMMY_REL(rel) && should_optimize_query())
+	Hypertable *ht;
+
+	if (prev_set_rel_pathlist_hook != NULL)
```

Comment:
So, since we're doing plan optimizations here, I think it is best to let potential other extensions do their planning first, in case our optimizations can work on top of those. For instance, if some other extension generates an Append path, we might want to optimize that to a ConstraintAwareAppend. I think this also applies to the sort optimization. Also note that this hook is not similar to other hooks that require a manual call to the "standard" function at the end.